### ISSUE TITLE
[Epic] Port all Backbone code to standard Django templates

### ISSUE THREAD

**Issue by brylie** [CiviWiki/OpenCiviWiki#1129]
### Idea summary

Many of our templates are defined with Backbone JavaScript code. However, we want to simplify our project and follow conventional Django development practices. To this end, we want to convert our Backbone code to Django templates.

### Further details

This task includes several sub-tasks, outlined below.

Note: this list is work-in-progress, pending some changes to the project folder structure.
- [x] #1153
- [x] #1151
- [ ] #1152

**Comment by odinmay** [comment]
Interested in working on this, could you assign it to me? Ill take a look at it

**Comment by brylie** [comment]
@odinmay, would you help us clarify the scope of work here? E.g., we need to list all of the Backbone templates that need to be ported over to Django template syntax. Could you help by creating that list as a comment on this issue?

**Comment by odinmay** [comment]
@brylie Yeah I can look into it later today. I may need some help as well, this is one of my first contributions but I'm looking forward to trying my best to help

**Comment by brylie** [comment]
Ok, sounds good. Thanks, @odinmay.

We'll learn about what code changes are needed by looking through the existing code structure. Once we identify a list of Backbone templates, we can divide and prioritize the work.

**Comment by AbhijithGanesh** [comment]
Can I be assigned to this along with @odinmay to reduce workload ?

**Comment by brylie** [comment]
@AbhijithGanesh, would you likevto help make the initial list of Backbone templates? Once we have the initial list of Backbone templates, the work can be divided up.

**Comment by AbhijithGanesh** [comment]
I'll develop the list and update it here asap.

**Comment by AbhijithGanesh** [comment]
We have about 26 templates, they're as follows:
-

-  account.html
- \accounts\templates\accounts\account.html:
- \accounts\templates\accounts\feed.html:
- \accounts\templates\accounts\password_reset.html:
- \accounts\templates\accounts\reset_by_email.html:
- \accounts\templates\accounts\settings.html:
- \accounts\templates\accounts\user-setup.html:
- \accounts\templates\accounts\partials\account\account_base.html:
- \accounts\templates\accounts\partials\account\account_settings.html:
- \accounts\templates\accounts\partials\account\login.html:
- \accounts\templates\accounts\register\entry_base.html:
- \accounts\templates\accounts\register\login.html:
- \accounts\templates\accounts\register\register.html:
- \accounts\templates\accounts\users\password_reset_complete.html:
- \accounts\templates\accounts\users\password_reset_confirm.html:
- \accounts\templates\accounts\users\password_reset_done.html:
- \accounts\templates\accounts\users\password_reset.html:
- \accounts\templates\accounts\utils\update_settings.html:
- \core\templates\about.html:
- \core\templates\base.html:
- \core\templates\how_it_works.html:
- \core\templates\landing.html:
- \core\templates\support_us.html:
- \threads\templates\threads\general-message.html:
- \threads\templates\threads\thread.html:
- \threads\templates\threads\partials\utils\global_nav.html:
Out of these a couple of them use Django Templating syntax, we need to convert those which doesn't use Django-Syntax

**Comment by wassafshahzad** [comment]
@brylie  can you assign me this issue ?

**Comment by brylie** [comment]
@wassafshahzad sure. It is a significant issue that consists of two parts (#1151 and #1152). I would appreciate it if you would take the lead and work holistically as you have done previously.

As you probably already understand, we need to

- convert the Backbone.js code to Django template syntax
- ensure there are conventional Django (class-based) views serving the relevant endpoints
- possibly add multiple Django CBVs per app, depending on the complexity of the Backbone templates we are replacing

**Comment by kobistar** [comment]
Hello,
@brylie Can I assign issue #1151 ?

**Comment by hauschie** [comment]
Hi I saw that this was unassigned to the users that said they would contribute. Does this mean that the task was completed or is there still work that needs to be done for this? I am a first time contributor.

**Comment by kobistar** [comment]
Hi, I wanted to do this assignment but I didn't have time. So this task is
available.
